From: Gonzalo Ruiz <Gonzalo.Ruiz@digi.com>
Date: Wed, 23 Sep 2020 10:15:08 +0200
Subject: [PATCH] imx8m: soc.mak: preserve dtbs after build

On commit ab433440269bbd5383b47ee60957e9906cda0b3b a dtb preprocess
was introduced to have compatibility with old U-Boots.

As part of those changes, the content of variable $(dtbs) was deleted
at the end of the build.

Since we added the dtbs variable as an argument, building is causing the
original dtb file to be removed, and a second build is failing since the
file pointed at by the dtbs variable does no longer exist.

Remove the file removal to avoid this.

Signed-off-by: Gonzalo Ruiz <Gonzalo.Ruiz@digi.com>
Signed-off-by: Gabriel Valcazar <gabriel.valcazar@digi.com>
---
 iMX8M/soc.mak | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 4bfdc37..b6595bc 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -149,7 +149,7 @@ u-boot.itb: $(dtb)
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtb)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtb) > u-boot.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot.its u-boot.itb
-	@rm -f u-boot.its $(dtb)
+	@rm -f u-boot.its
 
 dtb_ddr3l = valddr3l.dtb
 $(dtb_ddr3l):
@@ -283,7 +283,6 @@ print_fit_hab: u-boot-nodtb.bin bl31.bin $(dtb)
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtb)
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) VERSION=$(VERSION) ./print_fit_hab.sh $(PRINT_FIT_HAB_OFFSET) $(dtb)
-	@rm -f $(dtb)
 
 print_fit_hab_ddr4: u-boot-nodtb.bin bl31.bin $(dtb_ddr4_evk)
 	./$(PAD_IMAGE) tee.bin
